workspace/workspace.go

```diff
@@ -113,9 +113,22 @@ func (ws Workspace) SolutionDir(s string) (string, error) {
 		if _, err := os.Lstat(path); os.IsNotExist(err) {
 			return "", err
 		}
-		if _, err := os.Lstat(filepath.Join(path, solutionFilename)); err == nil {
+		if err := checkMetadataFile(path); err == nil {
```

Comment:
I'm not quite comfortable with how this method not only checks the files, but also does the migration.

What do you think about inlining the checking logic into this function, so that it explicitly checks the metadata file path and returns if it's there, then checks the legacy metadata file path and returns if it's there.

Then, after we've found the solution directory, call a `workspace.MigrateLegacyMetadataFile()` function (or some such name), which is a noop if the metadata file isn't legacy.

Reply:
Yeah, I didn't like